sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TriggerExecutor.scala

```diff
@@ -83,6 +84,6 @@ case class ProcessingTimeExecutor(processingTime: ProcessingTime, clock: Clock =
    * an interval of `100 ms`, `nextBatchTime(nextBatchTime(0)) = 200` rather than `0`).
    */
   def nextBatchTime(now: Long): Long = {
-    now / intervalMs * intervalMs + intervalMs
+    if (intervalMs == 0) now else now / intervalMs * intervalMs + intervalMs
```

Comment:
the doc seems wrong btw, mind fixing it? `nextBatchTime(nextBatchTime(0)) = 100` or am I understanding it wrong?